src/FileInfo.cpp

```diff
@@ -91,8 +91,19 @@ FileInfo GetFileInfo(const char *path)
       flags |= FileInfo::kFlagReadOnly;
 
     result.m_Flags = flags;
+
     // Do not allow directories to expose real timestamps, as it's not reliable behaviour across platforms
-    result.m_Timestamp = (flags & FileInfo::kFlagDirectory) ? kDirectoryTimestamp : stbuf.st_mtime;
+    result.m_Timestamp = (flags & FileInfo::kFlagDirectory) ? kDirectoryTimestamp : 
+    // high-precision timestaps in stat struct is not standardized. Different system headers 
+    // use different conventions - or don't support it at all (windows).
+#if defined(TUNDRA_APPLE)
+        stbuf.st_mtimespec.tv_sec * 1000000000 + stbuf.st_mtimespec.tv_nsec;
+#elif defined(TUNDRA_UNIX)
+        stbuf.st_mtim.tv_sec * 1000000000 + stbuf.st_mtim.tv_nsec;
+#else
+        stbuf.st_mtime * 1000000000;
```

Comment:
Should we not also fix this on Windows? It seems very likely the same issue you want to solve on mac and linux also exist on windows?

Reply:
Yes we should also fix this on windows. From the PR description:

>On windows, we can also get higher precision timestamps, but not using the stat function - we'd have to use win32 APIs for that (GetFileInformationByHandle/GetFileInformationByHandleEx). We would need to verify that these APIs are not slower than using stat.

Are you volunteering? :)

Reply:
Again, didn't read the description :P